test.js

```diff
@@ -0,0 +1,227 @@
+    // bits that relate to the cryptographic calculations.  We don't flip all
+    // bits for this public key since we're not doing signature file validation
+    // because of a self-signed certificate being used (i.g. -noverify)
+    it('should fail for bitflips on public key data', () => {
+      let pubkeydata = Buffer.from(pubkey).toString('ascii');
+      pubkeydata = pubkeydata.replace(/\n/g, '');
+      pubkeydata = pubkeydata.replace(/-----BEGIN CERTIFICATE-----/g, '');
+      pubkeydata = pubkeydata.replace(/-----END CERTIFICATE-----/g, '');
+      pubkeydata = Buffer.from(pubkeydata, 'base64');
+
+      const rsaStart = 289 + 4; // 286 is offset, 4 is header size
+      const rsaEnd = rsaStart + 257; // 257 is the size of the rsaEncryption
+      for (let i = rsaStart; i < rsaEnd; i++) {
+        for (let j = 0; j < 8; j++) {
+          let badPubkey = Buffer.from(pubkeydata);
+          badPubkey[i] ^= 1 << j;
+          badPubkey = Buffer.from('-----BEGIN CERTIFICATE-----\n'
+                                  + badPubkey.toString('base64')
+                                  + '\n-----END CERTIFICATE-----\n');
+          assume(subject(badPubkey, document, pkcs7_with_header)).is.not.ok();
+        }
+      }
+    });
+
+    // We're going to flip some bits here.  We're only going to try to flip the
+    // bits that relate to the cryptographic calculations.  We don't flip all
+    // bits for this signature since we're not doing signature file validation
+    // because of a self-signed certificate being used (i.g. -noverify)
+    it('should fail for bitflips on signature data', () => {
+      let x = new Array();
+      let sigdata = Buffer.from(pkcs7).toString('ascii');
+      sigdata = Buffer.from(sigdata, 'base64');
+
+      // The values here are obtained by using an ASN.1 decoder to find the
+      // cryptograpic parts of the signature to alter https://lapo.it/asn1js
+      const mdStart = 737 + 2; // 737 is offset, 2 is header size
+      const mdEnd = mdStart + 32; // 32 is the size of the messageDigest
+      const rsaStart = 786 + 4; // 786 is offset, 4 is header size
+      const rsaEnd = rsaStart + 256; // 256 is the size of the rsaEncryption
+      for (let i = mdStart; i < rsaEnd; i++) {
+        if (i > mdEnd && i < rsaStart) {
+          continue;
+        }
+        for (let j = 0; j < 8; j++) {
+          let badP7 = Buffer.from(sigdata);
+          badP7[i] ^= 1 << j;
+          badP7 = Buffer.from('-----BEGIN PKCS7-----\n'
+                              + badP7.toString('base64')
+                              + '\n-----END PKCS7-----\n');
+          try {
+            assume(subject(pubkey, document, badP7)).is.not.ok();
+          } catch (err) {
+            // We want to account for errors caused by our own generation of
+            // and invalid PEM file
+            let asn1Errors = [
+              /ASN1_CHECK_TLEN/,
+              /ASN1_ITEM_EX_D2I/
```

Comment:
nit: eslint was adding a trailing comma for me here.

Reply:
That's the taskcluster coding style, but I'll commit that fix